src/jit/flowgraph.cpp

```diff
@@ -25694,3 +25694,211 @@ bool Compiler::fgUseThrowHelperBlocks()
 {
     return !opts.compDbgCode;
 }
+
+//------------------------------------------------------------------------
+// fgTailMergeThrows: Tail merge throw blocks and blocks with no return calls.
+//
+// Notes:
+//    Scans the flow graph for throw blocks and blocks with no return calls
+//    that can be merged, and opportunistically merges them.
+//
+//    Does not handle throws yet as the analysis is more costly and less
+//    likely to pay off.
+void Compiler::fgTailMergeThrows()
+{
+    JITDUMP("\n*************** In fgTailMergeThrows\n");
+
+    if (opts.MinOpts())
+    {
+        JITDUMP("Method compiled with minOpts, no merging.\n");
+        return;
+    }
+
+    if (opts.compDbgCode)
+    {
+        JITDUMP("Method compiled with debug codegen, no merging.\n");
+        return;
+    }
+
+    struct ThrowHelperKey
+    {
+        BasicBlock*  m_block;
+        GenTreeCall* m_call;
+
+        ThrowHelperKey() : m_block(nullptr), m_call(nullptr)
+        {
+        }
+        ThrowHelperKey(BasicBlock* block, GenTreeCall* call) : m_block(block), m_call(call)
+        {
+        }
+    };
+
+    struct ThrowHelperKeyFuncs
+    {
+    public:
+        static bool Equals(const ThrowHelperKey& x, const ThrowHelperKey& y)
```

Comment:
These 2 can be moved to `ThrowHelperKey` to avoid the extra noise from a separate class.

Reply:
Merged the classes.